### ISSUE TITLE
random "INTERNAL_ERROR"

### ISSUE THREAD

**Issue by mrobertseidowsky-gpsw** [nak/pytest_mproc#2]
**Environnement**: Ubuntu 20.4, 12cores (6 multi-thread)
**Protocol**:
- The `setup.cfg` file contains:
```
[tool:pytest]
# Specify use of all available cores
addopts = --cores auto
```

**Observation**:
Sometimes I've got randomly these kind of error:
```
collecting ... INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
```

@nak  Do you have an idea of what happened and how can we avoid that ?

**Comment by nak** [comment]
I have a good idea of what might be going on. Some environments use local
host as the correct way to reference local connections but others require
127.0.0.1. it is an odd kind of thing. Let me see what I can do. I will try
to set up a fallback from one to the other in case the first one fails. I
can most likely get the work done over the weekend possibly sooner I hope
that works for you

**Comment by nak** [comment]
When you say random -- do you mean this happens intermittently?

Do you have full output from the run?

**Comment by mrobertseidowsky-gpsw** [comment]
> When you say random -- do you mean this happens intermittently?
> 

yes without specific context

> Do you have full output from the run?

I've got only that:
```
collecting ... INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
INTERNAL_ERROR> Exception in worker process: [Errno 111] Connection refused
INTERNAL_ERROR> not enough values to unpack (expected 5, got 4)
Waiting for server...
Connected
```
And after that the tests are correctly running, it doesn't make the program crash, it's just an error that we would like to avoid. Thanks in advance!

**Comment by nak** [comment]
I think this is resolved with 3.2.10.  Please let me know if not.  I cannot reproduce, but did add better error-checking and retries.

**Comment by mrobertseidowsky-gpsw** [comment]
I didn't see the error again in 3.2.10 thks!
